### DISCUSSION TITLE
synology reverse proxy issue

### DISCUSSION THREAD

**Discussion by paysenyu** [teslamate-org/teslamate#2776] [Q&A]
I have a problem when I use reverse proxy to proxy teslamate.

<img width="601" alt="截屏2022-07-19 11 23 10" src="https://user-images.githubusercontent.com/51367628/179657717-16e8c476-2bdb-4682-bc8d-e1611d74b9a4.png">

For example my domain is  http://a.com:4000 
And such as picture, I proxy http > https  so 
my domain now is https://a.com:4002 

Now I can surf https://a.com:4002  but I can`t see map on the web .
such as 
<img width="1140" alt="截屏2022-07-19 11 25 50" src="https://user-images.githubusercontent.com/51367628/179658013-1a371ce0-1323-40ce-8d73-07ac36c072f5.png">

and every button can`t be clicked...

please help me to fix it ..

**Comment by paysenyu** [comment]
update ..
If I try http://a.com:4000/ ,  map is ok  like this
<img width="1014" alt="截屏2022-07-19 11 27 36" src="https://user-images.githubusercontent.com/51367628/179658197-9e881389-50d4-4dc7-8823-90cac6895585.png">

and every button like  Range,Language can be clicked..
<img width="1111" alt="截屏2022-07-19 11 28 24" src="https://user-images.githubusercontent.com/51367628/179658284-2f5c383f-a07e-45c5-adc6-0bbd7a1099f8.png">

**Reply by sjayanna01** [reply]
Did you manage to fix this? Stuck with the same issue.

**Reply by sjayanna01** [reply]
Here's the fix.
Enable websockets by adding these 2 headers in the NAS reverse proxy settings.

![image](https://github.com/user-attachments/assets/459c41bd-ed2a-4fd8-90ab-0e80322db7dd)

Or below if you are running it on a docker container with caddy, nginx or something else.

            proxy_set_header Upgrade $http_upgrade;
            proxy_set_header Connection "upgrade";
            proxy_read_timeout 86400; //optional
